### ISSUE TITLE
Backport from SNN

### ISSUE THREAD

**Issue by alessandropellegrini** [ROOT-Sim/core#19]
Backport:
- [ ] Retractable
- [ ] Pubsub

**Comment by alessandropellegrini** [comment]
Depends on #21.
Assign `TAGS` in MODELINIT, which are dynamically assigned by the runtime.

**Comment by Piccions** [comment]
> @Piccions although still far from what we discussed, we should consider `retractable` as addressed by [282ad66](https://github.com/ROOT-Sim/core/commit/282ad6628c72e47d097ce3de6a5928ea6226097c) couldn't we?

For the purposes of SNNs and simple ABM models, that retractable module is _enough_. I'll write some tests and clean the code a bit. I'll open a merge request in the next days.

**Comment by alessandropellegrini** [comment]
Good! I'm marking `retractable` as done. The new `ScheduleNewEvent` interface will deserve a new issue.

**Comment by alessandropellegrini** [comment]
@AdrianoPi I think the backport of the pubsub module will still take some time, won't it? Shall we move this to v3.1.0?

**Comment by AdrianoPi** [comment]
It likely will. We can move it to v3.1.0 which might make things smoother.
